### PR TITLE
Adds a small chance to be a smartass when affected by Voice of God

### PR DESCRIPTION
**Pull request by Momo8289**
## About The Pull Request
Adds a small chance to be a smartass while following the "say my name" VOG command. 
## Why It's Good For The Game
It's funny :+1:
## Changelog
:cl:
add: There's now a small chance to be a smartass when affected by Voice of God
/:cl:

### CODE REVIEW

File: code/datums/voice_of_god_command.dm [hunk 1]
```diff
@@ -296,7 +296,11 @@ GLOBAL_LIST_INIT(voice_of_god_commands, init_voice_of_god_commands())
 /datum/voice_of_god_command/say_my_name/execute(list/listeners, mob/living/user, power_multiplier = 1, message)
 	var/iteration = 1
 	for(var/mob/living/target as anything in listeners)
```
Comment (MrMelbert) [inline code/datums/voice_of_god_command.dm:298]:
```suggestion
	var/regex/smartass_regex = regex(@"^say my name[.!]*$")
	for(var/mob/living/target as anything in listeners)
```
We don't wanna make the same regex a million times

File: code/datums/voice_of_god_command.dm [hunk 2]
```diff
@@ -296,7 +296,11 @@ GLOBAL_LIST_INIT(voice_of_god_commands, init_voice_of_god_commands())
...
-		addtimer(CALLBACK(target, TYPE_PROC_REF(/atom/movable, say), user.name), 0.5 SECONDS * iteration)
+		var/to_say = user.name
+		// 1% chance to be a smartass
+		if(findtext(lowertext(message), regex(@"^say my name[.!]*$")) && !rand(100))
```
Comment (MrMelbert) [inline code/datums/voice_of_god_command.dm:301]:
```suggestion
		if(findtext(lowertext(message), smartass_regex) && !rand(100))
```

Comment (MrMelbert) [inline code/datums/voice_of_god_command.dm:301]:
```suggestion
		if(findtext(lowertext(message), regex(@"^say my name[.!]*$")) && prob(1))
```

**Comment by MrMelbert** [comment]
Also IMO should be like a 0.1% chance rather than 1% chance, 1% is a lot more common than you think

File: code/datums/voice_of_god_command.dm [hunk 3]
```diff
@@ -295,8 +295,13 @@ GLOBAL_LIST_INIT(voice_of_god_commands, init_voice_of_god_commands())
 
 /datum/voice_of_god_command/say_my_name/execute(list/listeners, mob/living/user, power_multiplier = 1, message)
 	var/iteration = 1
+	var/regex/smartass_regex = regex(@"^say my name[.!]*$")
 	for(var/mob/living/target as anything in listeners)
-		addtimer(CALLBACK(target, TYPE_PROC_REF(/atom/movable, say), user.name), 0.5 SECONDS * iteration)
+		var/to_say = user.name
+		// 1% chance to be a smartass
+		if(findtext(lowertext(message), smartass_regex) && prob(0.01))
```
Comment (Ghommie) [inline code/datums/voice_of_god_command.dm:302]: That's a one in ten thousands chance.
```suggestion
		if(findtext(lowertext(message), smartass_regex) && prob(1))
```

**Comment by Ghommie** [comment]
Melbert, the 0.01% is too low! It has an overral probability of it occuring at least one in twenty repetitions of roughly 0.2%.
Also, if you're actually around 20 other mobs, you probably won't even notice the joke with all the flood of chat messages they're causing.

**Comment by MrMelbert** [comment]
A probability of 1% across 10 trials = 65% chance of at least 1 success
A probability of 0.1% across 10 trials = 10% chance of at least 1 success
A probability of 0.01% across 10 trials = 1% chance of at least 1 success

If we wanna make this a funny rare meme then I don't think 0.01% chance is too low (unless my statistics are bad)

**Pull request merged**